### ISSUE TITLE
Migrations placed in a second location are applied when using the gradle client and not when using Spring Boot

### ISSUE THREAD

**Issue by giordy** [flyway/flyway#1448]
###### What version of Flyway are you using?
4.0.3

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin, SBT plugin, ANT tasks)
Spring Boot and Gradle

###### What database are you using (type & version)?
PostgreSQL 9.5.4

###### What operating system are you using?
Fedora 24

###### What did you do?
I integrated FlyWay in my project, both in Spring Boot and in Gradle.
I did so by adding
```
plugins {
    id "org.flywaydb.flyway" version "4.0.3"
}
```
and the dependency
```
compile "org.flywaydb:flyway-core:4.0.3"
```
in my `build.gradle` 

Then I configured flyway in application.yml like this for Spring Boot:
```
  flyway:
    locations: classpath:db/migration,classpath:db/testdata
```
and in gradle like this:

```
flyway {
        user = '*myuser*'
        password = '*mypaswd*'
        url = '*myjdbcurl*'
        locations = ['classpath:db/migration','classpath:db/testdata']
}
```

and in `src/main/resources` I have:

```
db
    migrations:
        V1__init.sql
        V2__init_data.sql
        V3__add_prop_clone_update_state.sql
    testdata:
        V999999__test_data.sql
```

###### What did you expect to see?
After running the gradle client with:
```
./gradlew flywayClean
./gradlew flywayMigrate
./gradlew flywayInfo
```
or running the app with:
```
./gradlew clean build
./gradlew bootRun
./gradlew flywayInfo
```
I was expecting to see in both cases this output:
```
:flywayInfo
+---------+-----------------------------+---------------------+---------+
| Version | Description                 | Installed on        | State   |
+---------+-----------------------------+---------------------+---------+
| 1       | init                        | 2016-10-07 15:44:38 | Success |
| 2       | init data                   | 2016-10-07 15:44:39 | Success |
| 3       | add prop clone update state | 2016-10-07 15:44:39 | Success |
| 999999  | test data                   | 2016-10-07 15:44:39 | Success |
+---------+-----------------------------+---------------------+---------+

```

###### What did you see instead?
When running with Gradle I get the expected output:
```
:flywayInfo
+---------+-----------------------------+---------------------+---------+
| Version | Description                 | Installed on        | State   |
+---------+-----------------------------+---------------------+---------+
| 1       | init                        | 2016-10-07 15:44:38 | Success |
| 2       | init data                   | 2016-10-07 15:44:39 | Success |
| 3       | add prop clone update state | 2016-10-07 15:44:39 | Success |
| 999999  | test data                   | 2016-10-07 15:44:39 | Success |
+---------+-----------------------------+---------------------+---------+
```

Instead when running the app I get:
```
:flywayInfo
+---------+-----------------------------+---------------------+---------+
| Version | Description                 | Installed on        | State   |
+---------+-----------------------------+---------------------+---------+
| 1       | init                        | 2016-10-07 15:46:22 | Success |
| 2       | init data                   | 2016-10-07 15:46:23 | Success |
| 3       | add prop clone update state | 2016-10-07 15:46:23 | Success |
| 999999  | test data                   |                     | Pending |
+---------+-----------------------------+---------------------+---------+
```
So basically the migration script in the second location is not applied.

**Comment by axelfontaine** [comment]
This is either an issue with Spring Boot or with your configuration.

**Comment by R-Sandor** [comment]
A little late here I know but I had the same issue. @axelfontaine was right that it was Spring configuration. 

When Spring with Test task gets the locations it uses:
`spring.flyway.locations=classpath:db/migration,classpath:db/dev`
